### PR TITLE
(889a) Add sentence information client

### PR DESCRIPTION
**Pull request by yndajas**
## Context

We need to access sentence information including type and various dates

## Changes in this PR

- Update `Person`/`Prisoner` types to include `bookingId` to enable sentence information requests
- Add new type
- Update paths
- Add sentence information client and sentence `findSentenceAndOffenceDetails` method

## Release checklist

[Release process documentation](../doc/how-to/perform-a-release.md)

As part of our continuous deployment strategy we must ensure that this work is
ready to be released once merged.

### Pre-merge

- [ ] There are changes required to the Accredited Programmes API for this change to work...
  - [ ] ... and they have been released to production already

### Post-merge

- [ ] This adds, extends or meaningfully modifies a feature...
  - [ ] ... and I have written or updated an end-to-end test for the happy path in the [Accredited Programmes E2E repo](https://github.com/ministryofjustice/hmpps-accredited-programmes-e2e)
- [ ] This makes new expectations of the API...
  - [ ] ... and I have notified the API developer(s) of changes to the contract tests (Pact), or the API is already compliant
- [ ] [Manually approve](../doc/how-to/perform-a-release.md#releasing-to-the-preprod-environment) release to preprod
- [ ] [Manually approve](../doc/how-to/perform-a-release.md#releasing-to-the-production-environment) release to prod

### CODE REVIEW

File: server/@types/prisonApi/index.d.ts [hunk 1]
```diff
@@ -1,7 +1,14 @@
...
+type Caseload = {
   caseLoadId: string // eslint-disable-next-line @typescript-eslint/member-ordering
   caseloadFunction: string
   currentlyActive: boolean
   description: string
   type: string
 }
+
+type SentenceAndOffenceDetails = {
+  sentenceStartDate: string
```
Comment (yndajas) [inline server/@types/prisonApi/index.d.ts:10]: `sentenceDate`

File: server/testutils/factories/person.ts [hunk 2]
```diff
@@ -8,6 +8,7 @@ export default Factory.define<Person>(({ params }) => {
   const currentPrison = params.currentPrison || `${county} (HMP)`
 
   return {
+    bookingId: faker.string.numeric({ length: 10 }),
```
Comment (Gweaton) [inline server/testutils/factories/person.ts:11]: I'm not sure if these are sequential, but the example booking IDs appear to have a length of 6, might be worth updating here, and in the `prisoner` factory. They also come back as an integer so it's probably worth matching that on our own models too.

**Review by Gweaton — COMMENTED** [review]
Looks good - just a small thing to fix re: return types from the API.

There's a small typo in the "Add sentence information client" commit (`would'e`) but only if you're rebasing anyway!

File: server/testutils/factories/person.ts (hunk 2 above)
Comment (Gweaton) [reply server/testutils/factories/person.ts:11]: My mistake, this is a string (it's an integer in the Prison API for some reason)

**Review by Gweaton — APPROVED** [review]
Oops, sorry thought I'd approved this

**Comment by yndajas** [comment]
[Add sentence details to view referral page (L)](https://trello.com/c/ZaShNyr4/889-add-sentence-details-to-view-referral-page-l)

**Pull request merged**